File: labs-python/lab01.py
jog_sheila = input() # Jogada da Sheila.
jog_reginaldo = input() # Jogada do Reginaldo.

# Regras do jogo em um dict, cada chave contém o indice da jogada que derrota.
jogo = {
    "pedra" : [2,3],
    "papel" : [0,4],
    "tesoura" : [1,3],
    "lagarto" : [4,1],
    "spock" : [2,0]
    }

derrotaveis_reginaldo = [list(jogo.keys())[i] for i in jogo[jog_reginaldo]] # Lista dos nomes das jogadas derrotaveis por Reginaldo.
derrotaveis_sheila = [list(jogo.keys())[i] for i in jogo[jog_sheila]] # Lista dos nomes das jogadas derrotaveis por Sheila.

resultado = "empate" # Valor a ser imprimido pelo programa. Caso nenhum dos jogadores vença, valor permanece inalterado.

if(jog_sheila in derrotaveis_reginaldo): # Se a jogada de Sheila for derrotavel por Reginaldo...
    resultado = "Jornada nas Estrelas"
elif(jog_reginaldo in derrotaveis_sheila): # Se a jogada de Reginaldo for derrotavel por Sheila...
    resultado = "Interestelar"

print(resultado) #Imprima o valor de resultado

File: labs-cpp/lab01.cpp
#include <iostream>

using namespace std;

int get_play_index(string input)
{
    if(input == "pedra") return 0;
    if(input == "papel") return 1;
    if(input == "tesoura") return 2;
    if(input == "lagarto") return 3;
    if(input == "spock") return 4;

   return -1; 
}

int get_winner(int play_1, int play_2)
{
    int counters[5][2] = {{2,3},{0,4},{1,3},{4,1},{2,0}};

    if (play_1 == play_2)
    {
        cout << "empate";
        return 0;
    }

    if(play_2 == counters[play_1][0] || play_2 == counters[play_1][1])
    {
        cout << "Jornada nas Estrelas";
        return 0;
    }

    cout << "Interestelar";
    return 0;
}

int game()
{
    string input_1;
    string input_2;

    cin >> input_1;
    cin >> input_2;

    int play_1 = get_play_index(input_1);
    int play_2 = get_play_index(input_2);

    if (play_1 == -1 || play_2 == -1)
    {
        cout << "Jogadas inválidas!!!";
        return -1;
    }

    get_winner(play_1, play_2);

    return 0;
}

int main()
{
    game();

    return 0;
}